File: app/src/main/java/intent/ActVermelha.java
package intent;

import android.os.Bundle;
import android.widget.TextView;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import com.example.exemplo.R;

import java.util.Objects;

public class ActVermelha extends AppCompatActivity {

    String textoRecebido;

    TextView campoDeTexto;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.act_vermelha);
        if (getIntent().getExtras() != null) {
            textoRecebido = Objects.requireNonNull(getIntent().getExtras()).getString("chave");
            campoDeTexto = findViewById(R.id.textView);
            campoDeTexto.setText(textoRecebido);
        }

        ExemploLOL exemplo = new ExemploLOL();

        exemplo.patoFazQuack();

        ExemploLOLKt.noobReclamaDoLag();

        ExemploLOLKt.gatoMia("miau");
    }
}

File: app/src/main/java/intent/ActVermelha.kt
package intent

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.TextView
import com.example.exemplo.R

class ActVermelha : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.act_vermelha)
        intent.extras?.run {
            getString("chavoso")?.let {
                findViewById<TextView>(R.id.texticulo_0).text = it
            }
            getString("nat")?.let {
                findViewById<TextView>(R.id.texticulo_1).text = it
            }
            getString("vit")?.let {
                findViewById<TextView>(R.id.texticulo_2).text = it
            }
            getString("gun")?.let {
                findViewById<TextView>(R.id.texticulo_3).text = it
            }
        }
    }
}